File: test/aidbox_sdk/snapshots/dotnethl7-fhir-r4-core/MedicinalProductAuthorization.cs
using Aidbox.FHIR.Base;
using Aidbox.FHIR.Utils;

namespace Aidbox.FHIR.R4.Core;

public class MedicinalProductAuthorization : DomainResource, IResource
{
    public Base.Period? DataExclusivityPeriod { get; set; }
    public string? RestoreDate { get; set; }
    public Base.CodeableConcept[]? Jurisdiction { get; set; }
    public MedicinalProductAuthorization_JurisdictionalAuthorization[]? JurisdictionalAuthorization { get; set; }
    public MedicinalProductAuthorization_Procedure? Procedure { get; set; }
    public Base.CodeableConcept? LegalBasis { get; set; }
    public Base.Period? ValidityPeriod { get; set; }
    public Base.ResourceReference? Regulator { get; set; }
    public Base.CodeableConcept? Status { get; set; }
    public Base.Identifier[]? Identifier { get; set; }
    public string? StatusDate { get; set; }
    public string? DateOfFirstAuthorization { get; set; }
    public string? InternationalBirthDate { get; set; }
    public Base.ResourceReference? Holder { get; set; }
    public Base.CodeableConcept[]? Country { get; set; }
    public Base.ResourceReference? Subject { get; set; }

    public class MedicinalProductAuthorization_JurisdictionalAuthorization : BackboneElement
    {
        public Base.CodeableConcept? Country { get; set; }
        public Base.Identifier[]? Identifier { get; set; }
        public Base.CodeableConcept[]? Jurisdiction { get; set; }
        public Base.Period? ValidityPeriod { get; set; }
        public Base.CodeableConcept? LegalStatusOfSupply { get; set; }
    }

    public class MedicinalProductAuthorization_Procedure : BackboneElement
    {
        public object? Date    
        {
            get
            {
                if (DatePeriod is not null)
                {
                    return DatePeriod;
                }
        
                if (DateDateTime is not null)
                {
                    return DateDateTime;
                }
        
                return null;
            }
        
            set
            {
                if (value?.GetType() == typeof(Base.Period))
                {
                    DatePeriod = (Base.Period)value;
                    return;
                }
        
                if (value?.GetType() == typeof(string))
                {
                    DateDateTime = (string)value;
                    return;
                }
        
                throw new ArgumentException("Invalid type provided");
            }
        }
        public required Base.CodeableConcept Type { get; set; }
        public Base.Period? DatePeriod { get; set; }
        public Base.Identifier? Identifier { get; set; }
        public Base.ResourceReference[]? Application { get; set; }
        public string? DateDateTime { get; set; }
    }
}

File: test/aidbox_sdk/snapshots/python/hl7_fhir_r4_core/MedicinalProductAuthorization.py
from typing import Optional, List
from dataclasses import dataclass, field
from base import Period
from base import CodeableConcept
from base import DomainResource
from base import Reference
from base import Identifier
from base import BackboneElement

@dataclass(kw_only=True)
class MedicinalProductAuthorizationJurisdictionalAuthorization(BackboneElement):
    country: Optional[CodeableConcept] = None
    identifier: Optional[List[Identifier]] = field(default_factory=list)
    jurisdiction: Optional[List[CodeableConcept]] = field(default_factory=list)
    validity_period: Optional[Period] = None
    legal_status_of_supply: Optional[CodeableConcept] = None

@dataclass(kw_only=True)
class MedicinalProductAuthorizationProcedure(BackboneElement):
    type: CodeableConcept
    date_period: Optional[Period] = None
    identifier: Optional[Identifier] = None
    application: Optional[List[Reference]] = field(default_factory=list)
    date_date_time: Optional[str] = None

@dataclass(kw_only=True)
class MedicinalProductAuthorization(DomainResource):
    data_exclusivity_period: Optional[Period] = None
    restore_date: Optional[str] = None
    jurisdiction: Optional[List[CodeableConcept]] = field(default_factory=list)
    jurisdictional_authorization: Optional[List[MedicinalProductAuthorizationJurisdictionalAuthorization]] = field(default_factory=list)
    procedure: Optional[MedicinalProductAuthorizationProcedure] = None
    legal_basis: Optional[CodeableConcept] = None
    validity_period: Optional[Period] = None
    regulator: Optional[Reference] = None
    status: Optional[CodeableConcept] = None
    identifier: Optional[List[Identifier]] = field(default_factory=list)
    status_date: Optional[str] = None
    date_of_first_authorization: Optional[str] = None
    international_birth_date: Optional[str] = None
    holder: Optional[Reference] = None
    country: Optional[List[CodeableConcept]] = field(default_factory=list)
    subject: Optional[Reference] = None